client-core/src/synchronizer/auto_sync_data.rs

```diff
@@ -20,6 +20,10 @@ pub struct AutoSyncData {
     pub max_height: u64,
     /// current syncing wallet name
     pub wallet: String,
+    /// connected via websocket
+    pub connected: bool,
+    /// sync state : getting block, waiting, etc.
+    pub state: String,
```

Comment:
what if `WebsocketState` is extended with one more variant "Disconnected", and here it could just be:
```
pub state: WebsocketState
```

instead of bool+String?

Reply:
And add `#[derive(Serialize)]` to `WebsocketState`

Reply:
state is for core.
so that can be

connect_state: ConnectState, <- connecting, connected, disconnected
state: CoreState  <- get blocks, waiting, ready

Reply:
Core state only applies if it's connected, so something like:
```
pub enum ConnectionState {
  Connecting,
  Connected(CoreState),
  Disconnected
}
```
?

Reply:
ok , got it